src/gmres_e.m

```diff
@@ -0,0 +1,426 @@
+    %               Computational time in seconds.
+    %
+    %   References:
+    %   -----------
+    %
+    %   [1] Morgan, R. B. (1995). A restarted GMRES method augmented
+    %   with eigenvectors. SIAM Journal on Matrix Analysis and
+    %   Applications, 16(4), 1154-1171.
+    %
+    %   Copyright:
+    %   ----------
+    %
+    %   This file is part of the KrySBAS MATLAB Toolbox.
+    %
+    %   Copyright 2023 CC&MA - NIDTec - FP - UNA
+    %
+    %   KrySBAS is free software: you can redistribute it and/or modify it under
+    %   the terms of the GNU General Public License as published by the Free
+    %   Software Foundation, either version 3 of the License, or (at your
+    %   option) any later version.
+    %
+    %   KrySBAS is distributed in the hope that it will be useful, but WITHOUT
+    %   ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    %   FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+    %   for more details.
+    %
+    %   You should have received a copy of the GNU General Public License along
+    %   with this file.  If not, see <http://www.gnu.org/licenses/>.
+    %
+
+    % ----> Sanity check on the number of input parameters
+    if nargin < 2
+        error("Too few input parameters. Expected at least A and b.");
+    elseif nargin > 7
+        error("Too many input parameters.");
+    end
+
+    % ----> Sanity checks on matrix A
+    % Check whether A is non-empty
+    if isempty(A)
+        error("Matrix A cannot be empty.");
+    end
+
+    % Check whether A is square
+    [rowsA, colsA] = size(A);
+    if rowsA ~= colsA
+        error("Matrix A must be square.");
+    end
+
+    n = rowsA;
+    clear rowsA colsA;
+
+    % ----> Sanity checks on vector b
+    % Check whether b is non-empty
+    if isempty(b)
+        error("Vector b cannot be empty.");
+    end
+
+    % Check whether b is a column vector
+    [rowsb, colsb] = size(b);
+    if colsb ~= 1
+        error("Vector b must be a column vector.");
+    end
+
+    % Check whether b has the same number of rows as b
+    if rowsb ~= n
+        error("Dimension mismatch between matrix A and vector b.");
+    end
+
+    clear rowsb colsb;
+
+    % Special sanity checks for LGMRES here
+
+    % ----> Default value and sanityu checks for m
+    if (nargin < 3) || isempty(m)
+        m = min(n, 10);
+    end
+
+    % ----> If m > n, error message is printed
+    if m > n
+        error("m must satisfy: 1 <= m <= n.");
+    end
+
+    % ----> If m == n, built-in unrestarted gmres will be used
+    if m == n
+        warning("Full GMRES will be used.");
+        tic();
+        [gmres_x, gmres_flag, ~, ~, resvec] = gmres(A, b);
+        time = toc();
+        x = gmres_x;
+        if gmres_flag == 0
+            flag = 1;
+        else
+            flag = 0;
+        end
+        relresvec = resvec ./ resvec(1, 1);
+        return
+    end
+
+    % ----> If m < n AND k == 0, built-in gmres(m) will be used
```

Comment:
I believe this should be mentioned in the docstring to alert the user. Could you add one sentence in the description of the `k` parameter in L31?